version/src/lib.rs

```diff
@@ -27,27 +50,40 @@ fn compute_commit(sha1: Option<&'static str>) -> Option<u32> {
 
 impl Default for Version {
     fn default() -> Self {
+        let feature_set = u32::from_le_bytes(
+            solana_runtime::feature_set::ID.as_ref()[..4]
+                .try_into()
+                .unwrap(),
+        );
         Self {
             major: env!("CARGO_PKG_VERSION_MAJOR").parse().unwrap(),
             minor: env!("CARGO_PKG_VERSION_MINOR").parse().unwrap(),
             patch: env!("CARGO_PKG_VERSION_PATCH").parse().unwrap(),
             commit: compute_commit(option_env!("CI_COMMIT")),
+            feature_set,
         }
     }
 }
 
 impl fmt::Display for Version {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}.{}.{}", self.major, self.minor, self.patch,)
+    }
+}
+
+impl fmt::Debug for Version {
```

Comment:
@mvines btw, I'm going to tweak this a abit. Well, I miss `solana-XXX --version`'s commit hash output:

before:
```
$ solana-ledger-tool --version
solana-ledger-tool 1.4.0 775340c6

```

after:
```
$ solana-ledger-tool --version
solana-ledger-tool 1.4.0
```

Well, it's very useful for development purpose at least for me to know which build commit from bunch of laying binaries with `--version` and log output. Both has gone now.

How about this change?

- Add new macro `crate::full_version!` which `fmt("{:?}", Version)` and replace all over place in various `main.rs`.

or
 
- Add new macro `crate::simple_version!` and restore `crate::version!` and change  what you really wanted the `fmt("{}.{}.{}")` behavior.

Reply:
I've been lamenting it's removal as well. 

I think changing
https://github.com/solana-labs/solana/blob/b2fc7f7caa2061bd7eb3485175f2ebdcb299c053/version/src/lib.rs#L96
to `{:?}` will do the trick actually.

Reply:
+1 for commit hash 😁

Reply:
@mvines @CriesofCarrots just created #12636 :)